### PR TITLE
upgrade Django,django-registration,django-debug-toolbar and whitenoise

### LINKED ISSUE #7: Upgrade to the latest Django version
Which will break `django-registration` so either upgrade that to the latest or swap it out for `django-allauth`.

### PR DESCRIPTION
**Pull request by GorettiRivera**
closes #7 

I pulled develop branch and upgraded the below packages, the app ran without errors.

Django==3.1
django-debug-toolbar==3.1.1
django-registration==3.1.1
whitenoise==5.0.1

### CODE REVIEW

**Comment by bbelderbos** [comment]
@GorettiRivera how did you verify the app? 

This fails for me:
- rm -rf venv
- python3.8 -m venv venv && source venv/bin/activate
- pip install requirements.txt

```
(venv) [bobbelderbos@imac pbreadinglist_bbelderbos (develop)]$ python manage.py runserver
Performing system checks...

Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/utils/autoreload.py", line 53, in wrapper
    fn(*args, **kwargs)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/core/management/commands/runserver.py", line 118, in inner_run
    self.check(display_num_errors=True)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/core/management/base.py", line 392, in check
    all_issues = checks.run_checks(
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/core/checks/registry.py", line 70, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    return check_method()
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/urls/resolvers.py", line 408, in check
    for pattern in self.url_patterns:
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/urls/resolvers.py", line 589, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/urls/resolvers.py", line 582, in urlconf_module
    return import_module(self.urlconf_name)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/myreadinglist/urls.py", line 17, in <module>
    url(r'^accounts/', include('registration.backends.hmac.urls')),
  File "/Users/bobbelderbos/code/pbreadinglist_bbelderbos/venv/lib/python3.8/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'registration'
```

This is expected to break:

https://django-registration.readthedocs.io/en/3.1.1/upgrade.html

> The current release series of django-registration is the 3.x series, which is not backwards-compatible with the django-registration 2.x release series.

So you'd need to make one or more code changes to get this working.

Thanks,
Bob

**Comment by GorettiRivera** [comment]
Hi Bob, 

Please follow the below steps:

**Issue: "ModuleNotFoundError: No module named 'registration'"** 

1. 
copy django-registration code under pbreadinglist/
rename django-registration folder to registration

2. 
** settings.py **
add the 'registration' app under INSTALLED_APPS

3.
** urls.py ***
change this line 
from: 
`    url(r'^accounts/', include('registration.backends.hmac.urls')),`
to:
`  url(r'^accounts/', include('registration.backends.activation.urls')),`

ref: [https://django-registration.readthedocs.io/en/3.1.1/upgrade.html](url)

```
Renaming of two-step activation workflow
The two-step activation workflow was previously found at registration.backends.hmac; it has been renamed and is now found at registration.backends.activation.

```
**Issue: "WSGI application 'myreadinglist.wsgi.application' could not be loaded; Error importing module.** 

1. 
** wsgi.py ***
change the below lines:

from: 
```
from whitenoise.django import DjangoWhiteNoise
application = DjangoWhiteNoise(application)
```

to:
```
from whitenoise import WhiteNoise
application = WhiteNoise(application)

```

Now run: python manage.py runserver

```
(awesome-test) Gorettis-MacBook-Pro:pbreadinglist gorettirivera$ python manage.py runserver
Performing system checks...

System check identified no issues (0 silenced).
October 09, 2020 - 19:16:56
Django version 3.1, using settings 'myreadinglist.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.

```

Please note that I am also upgrading the below packages: 
Django==3.1
django-debug-toolbar==3.1.1
django-registration==3.1.1
whitenoise==5.0.1

Have pulled the changes to 'develop' branch. Let me know if you get any issue.

Regards
Goretti

File: myreadinglist/urls.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
     url(r'^books/', include('books.urls', namespace='books')),
     url(r'^users/(?P<username>.*)$', book_views.user_page, name='user_page'),
     url(r'^widget/(?P<username>.*)$', book_views.user_page_widget, name='user_page_widget'),
-    url(r'^accounts/', include('registration.backends.hmac.urls')),
+    url(r'^accounts/', include('registration.backends.activation.urls')),
```
[the remark this answers is not in the document]
Comment (bbelderbos) [reply myreadinglist/urls.py:17]: without this you get:

```
django.urls.exceptions.NoReverseMatch: Reverse for 'auth_login' not found. 'auth_login' is not a valid view function or pattern name.
```

File: registration/backends/activation/__init__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+"""
```
Comment (bbelderbos) [inline registration/backends/activation/__init__.py:1]: why did you pull the directory in? it's in your virtual environment when you pip install the module:

`venv/lib/python3.8/site-packages/django_registration`

File: myreadinglist/urls.py (hunk 1 above)
Comment (bbelderbos) [inline myreadinglist/urls.py:17]: After add that line I get:

`ModuleNotFoundError: No module named 'registration'`

We need the latest: `path('accounts/', include('django_registration.backends.activation.urls')),`
https://django-registration.readthedocs.io/en/3.1.1/quickstart.html

**Review by bbelderbos — CHANGES_REQUESTED** [review]
Sorry but pulling your branch this is not working

**Comment by bbelderbos** [comment]
This works for me: https://github.com/PyBites-Open-Source/pbreadinglist/pull/19/files

**Comment by bbelderbos** [comment]
@GorettiRivera I will merge in my working one for now closing this PR, but I did label it as "hacktoberfest-accepted" as per https://hacktoberfest.digitalocean.com/hacktoberfest-update so that you do get credit.

**Pull request closed without merging**